app/assets/javascripts/data_sets/editTable.js.coffee

```diff
@@ -1,29 +1,155 @@
-IS.onReady "data_sets/edit", ->
-  setupEditTable()
-
-  if namespace.controller is "data_sets" and namespace.action is "edit"
-    settings =
-      buttons: ['close', 'add', 'save']
-      bootstrapify: true
-      upload:
-        ajaxify: true
-        url: window.location.pathname.substr(0, window.location.pathname.lastIndexOf("/")) + "/edit"
-        success: (data, textStatus, jqXHR) ->
-          redirect = window.location.pathname.split "/"
-          redirect.pop()
-          url = redirect.join "/"
-          window.location = url
-          
-    ($ '#editTable').editTable(settings)
-    
-    field_count = ($ '#editTable').find('tr').eq(0).find('th').size() - 1
-    
-    ($ '#editTable').find('tr').slice(1).each () ->
-      ($ @).find('td:not(:last)').each () ->
-        ($ @).attr 'width', "#{95 / (field_count)}%"
-    ($ '#edit_table_add').click ->
-      ($ @).find('td:not(:last)').each () ->
-        ($ @).attr 'width', "#{95 / (field_count)}%"
-
-IS.onReady "data_sets/manualEntry", ->
-  setupEditTable()
+uploadSettings =
+  dataType: 'JSON'
+  urlEdit: '#'
+  urlEntry: window.location.pathname.substr(0, window.location.pathname.lastIndexOf("/")) + "/jsonDataUpload"
+  methodEdit: 'PUT'
+  methodEntry: 'POST'
+  error: (jqXHR, textStatus, errorThrown) ->
+    ($ '.edit_table_add').removeClass 'disabled'
+    ($ '.edit_table_save').button 'reset'
+    alert "An upload error occurred."
+  successEdit: (data, textStatus, jqXHR) ->
+    window.location = data.redirect
+  successEntry: (data, textStatus, jqXHR) ->
+    window.location = data['displayURL']
+
+setupTable = (cols, data) ->
+  cols.push
+    id: 'del'
+    field: 'del'
+    name: ''
+    width: 0
+    formatter: (row, cell, value, columnDef, dataContext) ->
+      "<i class='fa fa-close slick-delete'></i>"
+
+  options =
+    autoEdit: true
+    editable: true
+    enableCellNavigation: true
+    enableColumnReorder: false
+    forceFitColumns: true
+    rowHeight: 35
+    syncColumnCellResize: true
+  for x, i in data
+    data[i]['id'] = i
+
+  view = new Slick.Data.DataView()
+  grid = new Slick.Grid '#slickgrid-container', view, cols, options
+
+  grid.onClick.subscribe (e, args) ->
+    cell = grid.getCellFromEvent e
+    if cell.cell == grid.getColumns().length - 1
```

Comment:
You don't necessarily need to go back and fix this, but generally is pretty in coffeescript if you use
```is```
instead of 
```===```